### PR TITLE
Plugin v1.3

### PR DESCRIPTION
**Pull request by tiopi**
Refactored code to fix leaking tags and breadcrumbs across different exceptions.

### CODE REVIEW

File: sentry_plugin/hooks/sentry_hook.py [hunk 1]
```diff
@@ -4,64 +4,87 @@
...
 
 SCOPE_TAGS = frozenset(("task_id", "dag_id", "execution_date", "operator"))
+SCOPE_CRUMBS = frozenset(
+    ("dag_id", "task_id", "execution_date", "state", "operator", "duration")
+)
 
 
 @provide_session
-def get_task_instance(task, execution_date, session=None):
+def get_task_instances(dag_id, task_ids, execution_date, session=None):
```
Comment (Zylphrex) [inline sentry_plugin/hooks/sentry_hook.py]: This is a backwards incompatible change, think about your versioning.

Comment (tiopi) [reply sentry_plugin/hooks/sentry_hook.py]: This plugin does not have any explicit version currently.

File: sentry_plugin/hooks/sentry_hook.py [hunk 2]
```diff
@@ -70,27 +93,20 @@ class SentryHook(BaseHook):
...
         ignore_logger("airflow.task")
+        ignore_logger("airflow.jobs.backfill_job.BackfillJob")
         executor_name = configuration.conf.get("core", "EXECUTOR")
 
         sentry_flask = FlaskIntegration()
+        integrations = [sentry_flask]
 
         if executor_name == "CeleryExecutor":
             from sentry_sdk.integrations.celery import CeleryIntegration
+            from sentry_sdk.integrations.tornado import TornadoIntegration
```
Comment (untitaker) [inline sentry_plugin/hooks/sentry_hook.py]: Do we use tornado exactly whenever we use celery?

Comment (tiopi) [reply sentry_plugin/hooks/sentry_hook.py]: It shows as a suggestion when I look at the celery errors. On top of that, airflow installs tornado along with Celery.

**Comment by untitaker** [comment]
I would suggest to unconditionally enable the tornado integration. It is
not related to celery

File: sentry_plugin/hooks/sentry_hook.py [hunk 3]
```diff
@@ -70,27 +93,20 @@ class SentryHook(BaseHook):
...
-            import logging
-            from sentry_sdk.integrations.logging import LoggingIntegration
-
-            sentry_logging = LoggingIntegration(
-                level=logging.INFO, event_level=logging.ERROR
-            )
-            integrations = [sentry_logging]
-
-        integrations.append(sentry_flask)
+            sentry_tornado = TornadoIntegration()
```
Comment (maxcountryman) [inline sentry_plugin/hooks/sentry_hook.py]: Do you want to add this to the integrations list below?

File: sentry_plugin/hooks/sentry_hook.py (hunk 2 above)
Comment (nathanielvarona) [reply sentry_plugin/hooks/sentry_hook.py]: Airflow's Celery doesn't use Tornado. It only uses by the add-on Celery monitoring tool called `Flower` so no need to integrate Tornado unless we need instrumentation too for Flower.

See:
https://docs.celeryproject.org/en/latest/userguide/monitoring.html#flower-real-time-celery-web-monitor
https://airflow.apache.org/cli.html#flower

File: sentry_plugin/hooks/sentry_hook.py (hunk 3 above)
Comment (tiopi) [reply sentry_plugin/hooks/sentry_hook.py]: The tornado has been causing some issues when I test celery on Google Composer. This is one reason why I haven not been able to merge this PR just yet. I will get this resolved by tomorrow latest.

Comment (maxcountryman) [reply sentry_plugin/hooks/sentry_hook.py]: Curious what issues you've been running into? I've been running this on Heroku and things seem to be working without any major issues.

Comment (tiopi) [reply sentry_plugin/hooks/sentry_hook.py]: Tornado requires Python3.6+ or the aiocontextvars package. For some reason this is creating an issue with using the `sentry_dsn` from the connections database, and the exceptions are not sent to Sentry.

**Pull request merged**